openvdb_houdini/SOP_OpenVDB_Points_Convert.cc

```diff
@@ -799,6 +799,45 @@ SOP_OpenVDB_Points_Convert::Cache::cookVDBSop(OP_Context& context)
         gdp->clearAndDestroy();
 
         const GU_Detail* ptGeo = inputGeo(0, context);
+        GU_Detail nonConstDetail; // tmp storage of unpacked geo
+        const GU_Detail* detail;  // ptr to geo to convert; either ptGeo or nonConstDetail
+
+        boss.start();
+
+        // Unpack any packed primitives
+
+        std::unique_ptr<GA_PrimitiveGroup> packgroup(ptGeo->newDetachedPrimitiveGroup());
+
+        for (GA_Iterator it(ptGeo->getPrimitiveRange()); !it.atEnd(); ++it) {
+            GA_Offset offset = *it;
+            const GA_Primitive* primitive = ptGeo->getPrimitive(offset);
+            if (!primitive || !GU_PrimPacked::isPackedPrimitive(*primitive)) continue;
+
+            const GU_PrimPacked* packedPrimitive = static_cast<const GU_PrimPacked*>(primitive);
+            packedPrimitive->unpack(nonConstDetail);
+            packgroup->addOffset(offset);
+        }
+
+        if (packgroup->entries() == 0) {
+            // If no packed geometry was converted, avoid the merge by
+            // simply using the original geometry
+            detail = ptGeo;
+        }
+        else {
+            // Convert the prim group to points - we have to use mergePoints
+            // instead of mergePrims to make sure we merge points that are not
+            // associated with any primitive
+            GA_PointGroup pointsWithPackedPrims(*ptGeo);
+            pointsWithPackedPrims.GA_ElementGroup::operator|=(*packgroup);
+
+            // Merge everything except the geo associated with prims we've just unpacked
+            nonConstDetail.mergePoints(*ptGeo, GA_Range(pointsWithPackedPrims, /*invert*/true));
```

Comment:
@jmlait might be worth removing the operator overrides for `GA_PointGroup`, `GA_PrimitiveGroup` etc as they seem to just call through to `GA_ElementGroup` and obfuscate the input group type - assuming I've understood this behaviour correctly!

Reply:
The &=, -=, etc overrides make a lot of sense as you can only apply that with matching group types.  So you can't have 
pointgroup -= primgroup;
As far as I understand, the overrides then enforce that restriction.
The issue is that |= does implicit conversion of the group type.  The ElementGroup version used to be a private method until GA_ElementGroup was merged into GA_ATIGroupBool. The latter was always meant to be a private interface, but got exposed as ElementGroup must be public.  (This was 4y9m ago, so likely compatible with all current OpenVDB targets)

Long story short, the |= working with different group types wasn't meant to be exposed :>  I think the best solution is to have the GA_PointGroup overrides accept an ElementGroup as the second input.  I'll add an RFE to this effect and see if there are any issues.

Reply:
Why not use combine() rather than this awkward |=?

Reply:
I wasn't entirely sure if combine did the same thing but will definitely switch if so! Is this the main way to "cast" one group type to another?

Reply:
Yep, combine will chain to |= if the src is an element group. It then handles in addition weird stuff like edge groups.

The |= in ElementGroup will be marked deprecated in 18.5.

Reply:
Switch to combine!